frontend/src/component/layout/MainLayout/MainLayout.tsx

```diff
@@ -90,14 +92,20 @@ const MainLayoutContentContainer = styled('main')(({ theme }) => ({
 
 export const MainLayout = forwardRef<HTMLDivElement, IMainLayoutProps>(
     ({ children }, ref) => {
+        const newAdminUIEnabled = useUiFlag('adminNavUI');
         const { uiConfig } = useUiConfig();
+        const location = useLocation();
         const projectId = useOptionalPathParam('projectId');
         const { isChangeRequestConfiguredInAnyEnv } = useChangeRequestsEnabled(
             projectId || '',
         );
 
         const theme = useTheme();
         const isSmallScreen = useMediaQuery(theme.breakpoints.down('lg'));
+        const showOnlyAdminMenu =
+            newAdminUIEnabled && location.pathname.indexOf('/admin') === 0;
```

Comment:
will it break with a route to feature flag named admin?

Reply:
it shouldn't, but I decided to test:
![image](https://github.com/user-attachments/assets/cf8c8fb7-2892-48fe-9711-ed37b2475ed3)